src/release/breaking-changes/context-menus.md

```diff
@@ -0,0 +1,328 @@
+    );
+  },
+)
+```
+
+A large number of examples of different custom context menus are available [in
+the samples repo](https://github.com/flutter/samples/tree/main/experimental/context_menus).
+
+All related deprecated features were flagged with the deprecation warning "Use
+`contextMenuBuilder` instead."
+
+## Migration guide
+In general, any previous changes to context menus that have been deprecated now
+require the use of the contextMenuBuilder parameter on the relevant text-editing
+or text-selection widget (
+[on TextField](https://api.flutter.dev/flutter/material/TextField/contextMenuBuilder.html),
+for example). Return a built-in context menu widget like
+[AdaptiveTextSelectionToolbar](https://api.flutter.dev/flutter/material/AdaptiveTextSelectionToolbar-class.html)
+to use Flutter's built-in context menus, or return your own widget for something
+totally custom.
+
+To transition to contextMenuBuilder, the following parameters and classes have
+been deprecated.
+
+### [ToolbarOptions](https://api.flutter.dev/flutter/widgets/ToolbarOptions-class.html)
+This class was previously used to explicitly enable or disable certain buttons
+in a context menu. Before this change, you might have passed it into TextField
+or other widgets like this:
+
+```dart
+// Deprecated.
+TextField(
+  toolbarOptions: ToolbarOptions(
+    copy: true,
+  ),
+)
+```
+
+Now, you can achieve the same effect by adjusting the buttonItems passed into
+AdaptiveTextSelectionToolbar. For example, you could ensure that the Cut button
+never appears, but the other buttons do appear as usual:
+
+```dart
+TextField(
+  contextMenuBuilder: (context, editableTextState) {
+    final List<ContextMenuButtonItem> buttonItems =
+        editableTextState.contextMenuButtonItems;
+    buttonItems.removeWhere((ContextMenuButtonItem buttonItem) {
+      return buttonItem.type == ContextMenuButtonType.cut;
+    });
+    return AdaptiveTextSelectionToolbar.buttonItems(
+      anchors: editableTextState.contextMenuAnchors,
+      buttonItems: buttonItems,
+    );
+  },
+)
+```
+
+Or, you could ensure that the Cut button appears exclusively and always:
+
+```dart
+TextField(
+  contextMenuBuilder: (context, editableTextState) {
+    return AdaptiveTextSelectionToolbar.buttonItems(
+      anchors: editableTextState.contextMenuAnchors,
+      buttonItems: <ContextMenuButtonItem>[
+        ContextMenuButtonItem(
+          onPressed: () {
+            editableTextState.cutSelection(SelectionChangedCause.toolbar);
+          },
+          type: ContextMenuButtonType.cut,
+        ),
+      ],
+    );
+  },
+)
+```
+
+### [TextSelectionControls.canCut](https://api.flutter.dev/flutter/widgets/TextSelectionControls/canCut.html) and other button booleans
+These booleans previously had the same effect of enabling and disabling certain
+buttons as ToolbarOptions.cut, etc. had. Before this change, you might have been
+hiding and showing buttons by overriding TextSelectionControls and setting these
```

Comment:
```suggestion
buttons as `ToolbarOptions.cut`, and so on had. Before this change, you might have been
hiding and showing buttons by overriding `TextSelectionControls` and setting these
```

Reply:
I was really watching my "i.e."s and "e.g."s, but I forgot about "etc." :)